### PR TITLE
Add card component

### LINKED ISSUE #21: Add Card component
WIP

References
- https://material.io/guidelines/components/cards.html
- https://getmdl.io/components/index.html
- http://www.material-ui.com/#/components/card

### PR DESCRIPTION
**Pull request by remon-georgy**
Fixes #21.

This is a minimal implementation of Card component, as well as sub-components (Title, Text and Actions).

Implementation goals are:
- Stick to the MD's non-controversial styling and layout guidelines.
- Leave complicated/conditional MD guidelines to be decided by end developers. For example, CardText special padding when it's followed by CardActions, that is not implemented.
- Adapt those guidelines to different screen sizes (using Uranium).
- Default style could be overridden.
- Rely on Carbon UI's packaged Types, Elevation and Colors as much as possible.

TODO these tasks could be tackled in following tickets
- Unit testing!!!
- Add documentation to carbon-ui-docs
- Implement Card elevation animations, e.g while hovering over a card, elevation should change to 8dp.
- Collapse/Expand feature.
- Have Card and its subcomponents act as [action areas](https://material.io/guidelines/components/cards.html#cards-actions) 

I'd appreciate all kinds of feedback to make this PR fit holistically in Carbon UI :).

### CODE REVIEW

**Comment by tuckerconnelly** [comment]
Dude you're a legend! I'm SUPER busy right now, but I'll try to find some time this weekend to look it over and give feedback. Thanks so much and stoked you're contributing :)

**Comment by remon-georgy** [comment]
Awesome! Your feedback would be great!

File: src/components/Card/Card.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import React, { Component, PropTypes } from 'react'
+import { View } from 'react-native'
+import Uranium from 'uranium'
+import { Elevation, Breakpoints, Colors, connectTheme } from '../../index'
+
+class Card extends Component {
```
Comment (tuckerconnelly) [inline src/components/Card/Card.js]: You can add a comment above this to automatically get the docs into carbon-ui.com, check other components for example

File: src/components/Card/Card.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    PropTypes.array,
+  ]),
+}
+
+const styles = {
+  root: {
+    marginTop: 4,
+    marginBottom: 4,
+    marginLeft: 8,
+    marginRight: 8,
```
Comment (tuckerconnelly) [inline src/components/Card/Card.js]: Nit, margin vertical/horizontal, more succinct :P

Also, I like ordering css props a la csslint, so another nit is, like breaks between here and the flex stuff below. Groupings go:

positioning
box model (width/height -> padding -> border -> margin)
flexbox
text
background
other stuff

with line breaks between all of them.

File: src/components/Card/CardText.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      PropTypes.array,
+    ]),
+    /**
+     * Text style.
+     */
+    textStyle: PropTypes.oneOfType([
+      PropTypes.object,
+      PropTypes.array,
+    ]),
+  };
```
Comment (tuckerconnelly) [inline src/components/Card/CardText.js]: Props at bottom of file, like `CardText.propTypes`--see other components for example

This is so, proptypes can be in the same part of the file for both functional and stateful components.

File: src/components/Card/CardText.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      paddingLeft: 24,
+      paddingRight: 24,
+    },
+  },
+  text: {
+    fontSize: 14,
+  },
+}
+
+export default connectTheme(Uranium(CardText))
```
Comment (tuckerconnelly) [inline src/components/Card/CardText.js]: Nit, line breaks to look like decorators, like in other components

File: src/components/Card/CardTitle.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+class CardTitle extends Component {
+  static propTypes = {
+    children: PropTypes.node,
+    subtitle: PropTypes.node,
+    subtitleStyle: PropTypes.object,
+    title: PropTypes.node,
+    titleStyle: PropTypes.object,
+    theme: PropTypes.object.isRequired,
+    style: PropTypes.oneOfType([
```
Comment (tuckerconnelly) [inline src/components/Card/CardTitle.js]: Man, I gotta write contributing guidelines, realizing how many OCD rules I have haha.

children/style at the bottom 'cause they're not as important. Props that are unique to this component go at the top 'cause they're the most important. Also, add comment if prop comes from a higher-order function. See other components

**Comment by remon-georgy** [comment]
Thanks @tuckerconnelly! Required changes are quite reasonable. Working on them.

P.S perhaps we can enforce those guidelines via additional ESLint rules?

**Comment by remon-georgy** [comment]
I can't believe it took me  weeks to get back to the PR. Anyway, all requested changes have been applied!

**Comment by remon-georgy** [comment]
Hi @tuckerconnelly, is there anything left before merging this component? I'm happy to apply any recomendations you might have.

**Comment by tuckerconnelly** [comment]
Coding wise looks rad. Gonna trust it's up to spec. Merging.

Btw, moving all my efforts over to https://github.com/next-md/next-md now--kinda in to next.js and PWAs currently. Goal is to completely transfer over carbon-ui.

**Comment by remon-georgy** [comment]
Thanks for the heads-up about next-md, @tuckerconnelly. I hope it's as suitable as carbon-ui to my use case.

**Pull request merged**